sklearn/decomposition/kernel_pca.py

```diff
@@ -259,7 +262,7 @@ def inverse_transform(self, X):
         "Learning to Find Pre-Images", G BakIr et al, 2004.
         """
         if not self.fit_inverse_transform:
-            raise ValueError("Inverse transform was not fitted!")
+            raise NotFittedError("Inverse transform was not fitted.")
```

Comment:
I did this since inverse transform seems to employ a different [fit method](https://github.com/scikit-learn/scikit-learn/pull/4029/files#diff-d6cfec34cb25a8568bc721be1ce40e44R175) which in our this case was not called... not sure though... please advise accordingly...